src/Translator.php

```diff
@@ -101,77 +102,12 @@ class Translator
     /** @var int|null number of plurals */
     private $pluralCount = null;
 
-    /**
-     * Array with original -> translation mapping.
-     *
-     * @var array<string,string>
-     */
-    private $cacheTranslations = [];
+    /** @var CacheInterface */
+    private $cache;
 
-    /**
-     * @param string|null $filename Name of mo file to load (null to not load a file)
-     */
-    public function __construct(?string $filename)
+    public function __construct(CacheInterface $cache)
```

Comment:
What about still accepting the filename to avoid a BC break and only accept CacheInterface on next major?

```suggestion
    /**
     * @param CacheInterface|string|null $filenameOrCache
     */
    public function __construct($filenameOrCache)
```

Reply:
That would definitely help anyone using or packaging this library

Reply:
If zero BC breaks can be made the I would be grateful for that
It would allow me to package the new version in Debian without having to wait

Reply:
@williamdes I've updated the constructor to accept `CacheInterface|string|null`